### PR TITLE
MG-283: Align Disease Correlation results with ui_config columns

### PR DESCRIPTION
**Pull request by beatrizsaldana**
## Problem
The disease correlation ETL currently produces a nested data structure where correlation results are stored in a `results` array containing objects with `module`, `correlation`, and `adj_p_val` fields. This nested structure complicates the process of combining the payload with `ui_config` data, as it requires additional processing to flatten the structure and map module values to UI configuration column definitions.

**Current Structure:**
```json
{
  "model": "5xFAD (IU/Jax/Pitt)",
  "matched_control": "C57BL6J", 
  "model_type": "Familial AD",
  "modified_genes": ["App", "Psen1"],
  "cluster": "Consensus Cluster A (ECM organization)",
  "age": "12 months",
  "sex": "Female",
  "results": [
    {
      "module": "IFG",
      "correlation": 0.0970533422952035,
      "adj_p_val": 0.388719974861466
    },
    {
      "module": "PHG", 
      "correlation": 0.084295638183579,
      "adj_p_val": 0.409231075552908
    }
  ]
}
```

## Solution
Updated the disease correlation ETL to produce a flattened data structure by:

1. **Removing the nested `results` array** - Correlation data is now stored directly at the root level of each record
2. **Using module names as keys** - Each module (e.g., "IFG", "PHG", "TCX") becomes a top-level key containing the correlation and adjusted p-value data

**New Structure:**
```json
{
  "model": "5xFAD (IU/Jax/Pitt)",
  "matched_control": "C57BL6J",
  "model_type": "Familial AD", 
  "modified_genes": ["App", "Psen1"],
  "cluster": "Consensus Cluster A (ECM organization)",
  "age": "12 months",
  "sex": "Female",
  "IFG": {
    "correlation": 0.0970533422952035,
    "adj_p_val": 0.388719974861466
  },
  "PHG": {
    "correlation": 0.084295638183579,
    "adj_p_val": 0.409231075552908
  }
}
```

## Changes Made

**Core Implementation (`src/agoradatatools/etl/transform/disease_correlation.py`):**
- Modified `process_group()` function to directly assign module data as top-level keys instead of collecting in a `results` array
- Updated the output structure to use module name as keys for correlation data
- Maintained all existing validation and error handling logic

**Testing (`tests/transform/test_disease_correlation.py`):**
- Updated test assertions to verify the new flattened structure
- Ensured module names are correctly extracted and used as keys
- Verified that correlation and p-value data is properly nested under module keys
- Maintained comprehensive test coverage for edge cases and error conditions

**Test Assets (`tests/test_assets/disease_correlation/output/disease_correlation_expected_output.json`):**
- Updated expected output to reflect the new flattened structure
- Preserved all test data integrity while demonstrating the new format

## Testing
- All existing unit tests pass with updated assertions
- Test assets updated to reflect new output structure
- Comprehensive validation of module name extraction and data nesting
- Error handling and edge case coverage maintained

### CODE REVIEW

File: src/agoradatatools/process.py [hunk 1]
```diff
@@ -159,12 +159,11 @@ def process_dataset(
         entities_as_df[entity_name] = df
 
     if "custom_transformations" in dataset_obj[dataset_name].keys():
-        result = apply_custom_transformations(
+        df = apply_custom_transformations(
             datasets=entities_as_df,
             dataset_name=dataset_name,
             dataset_obj=dataset_obj[dataset_name],
         )
-        df = utils.convert_transformation_result_to_dataframe(result, dataset_name)
```
Comment (beatrizsaldana) [inline src/agoradatatools/process.py]: @linglp do we know why this function was added? We already have a `load.list_to_json()` function.

I removed this because it was adding 300 null values.

File: tests/transform/test_disease_correlation.py [hunk 2]
```diff
@@ -127,16 +126,15 @@ class TestTransformDiseaseCorrelation:
...
                     and entry["sex"] == "Female"
-                    and len(entry["results"]) == 2
-                    and entry["results"][0]["module"] == "IFG"
-                    and entry["results"][1]["module"] == "PHG"
-                    and isinstance(entry["results"][0]["correlation"], float)
-                    and isinstance(entry["results"][0]["adj_p_val"], float)
+                    and len(entry["IFG"]) == 2
+                    and len(entry["PHG"]) == 2
+                    and isinstance(entry["IFG"]["correlation"], float)
+                    and isinstance(entry["PHG"]["adj_p_val"], float)
```
Comment (jaclynbeck-sage) [inline tests/transform/test_disease_correlation.py]: I'm running into gaps in my ability to understand Python: Can you walk me through what this lambda function is doing?

Comment (jaclynbeck-sage) [reply tests/transform/test_disease_correlation.py]: Specifically, I understand that it's checking a bunch of fields, but I don't know what the `any()` combined with `and` statements is going to return. I also don't know why it's only checking IFG correlation and PHG adj_p_val, and not both values for both tissues.

Would it be simpler to just create an expected output dictionary with the correct structure/values and compare them?

File: tests/transform/test_disease_correlation.py [hunk 3]
```diff
@@ -173,16 +162,27 @@ class TestTransformDiseaseCorrelation:
...
+                        {"name": "LOAD1", "gene": "APOE4"},
                     ]
                 ),
             },
-            lambda output: (
-                len(output) == 1
-                and len(output[0]["results"]) == 2
-                and output[0]["results"][0] == output[0]["results"][1]
-            ),
+            lambda output: output[0]["modified_genes"] == "APOE4",
```
Comment (jaclynbeck-sage) [inline tests/transform/test_disease_correlation.py]: I think in general it would be really helpful to add comments to the test functions and to these lambdas explaining what exactly is being tested and what the expected output should be for non-routine things. For example, I see that you are adding a duplicate `allele_info` item, but how does looking at `output[0]["modified_genes"] == "APOE4"` confirm that it has handled the duplicate properly? I would expect, for example, another check to make sure there isn't a duplicated output item too, or something.

Comment (jaclynbeck-sage) [reply tests/transform/test_disease_correlation.py]: Or also, should we be throwing an error if there's a duplicate, like we see for the other test cases with duplicates in the various structures?

**Review by jaclynbeck-sage — CHANGES_REQUESTED** [review]
The main function changes look good, but I would like some more comments in the test functions to explain what is happening.

File: src/agoradatatools/process.py (hunk 1 above)
Comment (JessterB) [reply src/agoradatatools/process.py]: Probably related to https://sagebionetworks.jira.com/browse/DPE-1384

Comment (beatrizsaldana) [reply src/agoradatatools/process.py]: Addressed in #217

File: tests/transform/test_disease_correlation.py (hunk 2 above)
Comment (beatrizsaldana) [reply tests/transform/test_disease_correlation.py]: I created an  expected output dictionary with the correct structure/values to make this more readable :)

**Review by jaclynbeck-sage — APPROVED** [review]
Looks good! Thanks for adding all the comments!
